autoload/go/impl.vim

```diff
@@ -35,8 +35,14 @@ function! go#impl#Impl(...) abort
 
   let result = go#util#System(printf("%s '%s' '%s'", binpath, recv, iface))
   if go#util#ShellError() != 0
-    call go#util#EchoError(result)
-    return
+    call go#util#EchoWarning("impl failed.  trying with package path")
+
+    let s:dirname = fnameescape(expand('%:p:h'))
+    let result = go#util#System(printf("%s -dir '%s' '%s' '%s'", binpath, s:dirname, recv, iface))
+    if go#util#ShellError() != 0
+      call go#util#EchoError(result)
+      return
+    end
   endif
```

Comment:
* The current version of the `impl` command ignores `vendor` when finding the interface to generate.  It will fail if the interface is installed in `vendor` but not `$GOPATH`.
 * josharian/impl#18 adds `-dir <dir>` to root the normal go package resolution rules at `<dir>`, which fixes the `vendor` issue.  `<dir>` defaults to `$PWD`.
 * The current version of `impl` does not take a `-dir <dir>` argument.

Maybe an example will help.  `impl.new` is from josharian/impl#18 and `impl.orig` is the current release:

```
$ mkdir -p $GOPATH/src/test.com/user/vim-go-impl
$ cd $GOPATH/src/test.com/user/vim-go-impl
$ mkdir -p vendor/src/test.com/foo
$ cat > vendor/src/test.com/foo/foo.go <<EOF
> package foo
> type I interface {
>   Bar()
> }
> EOF
$ impl.orig 'i *I' foo.I
interface foo.I not found: couldn't find package src/test.com/foo: cannot find package "src/test.com/foo" in any of:
  /usr/local/Cellar/go/1.8.3/libexec/src/src/test.com/foo (from $GOROOT)
  /Users/abozanich/gopath/src/src/test.com/foo (from $GOPATH)
$ impl.new 'i *I' foo.I
func (i *I) Bar() {
  panic("not implemented")
}

$ cd ..
$ impl.new 'i *I' foo.I
unrecognized interface: foo.I
$ impl.new -dir $PWD/vim-go-impl 'i *I' foo.I
func (i *I) Bar() {
  panic("not implemented")
}

$ impl.orig -dir $PWD/vim-go-impl 'i *I' foo.I
impl <recv> <iface>

impl generates method stubs for recv to implement iface.

Examples:

impl 'f *File' io.Reader
impl Murmur hash.Hash

Don't forget the single quotes around the receiver type
to prevent shell globbing.
```
It runs another if it fails because I didn't want to break installs with an older `impl` build.  Omitting the `-dir <dir>` is usually ok for my workflow because I'm generally cd'd into the directory that has `vendor`.